jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -1456,6 +1456,13 @@ private DynamicCapacity(List<RetainableByteBuffer> buffers, ByteBufferPool.Sized
                 throw new IllegalArgumentException("must always retain if cannot aggregate");
         }
 
+        private void checkNotReleased()
```

Comment:
This method is only used in this class, but I guess all subclasses of `Abstract` would need to do the same checks.
Feels like it's either too much defensive programming, or it needs to be done in the base class once.

Reply:
If those checks were done in the base class, pretty much all method implementations would need to call super. This would basically double the changes and I don't see how this would improve clarity.